### PR TITLE
[Frameworks] Fail lint for Swift on iOS < 8.

### LINKED ISSUE #2963: [Frameworks] Let lint fail for Swift pods supporting deployment targets below iOS 8.0
If a podspecs includes Swift files, let the linter fail if the spec specifies a deployment targets on iOS below 8.0.

### PR DESCRIPTION
**Pull request by neonichu**
Have to use dynamic frameworks for Swift and they are unsupported on iOS 7.

Fixes #2963

### CODE REVIEW

**Comment by orta** [comment]
rockin

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -310,6 +310,10 @@ def install_pod
       installer.install!
 
       file_accessors = installer.aggregate_targets.map do |target|
+        if target.pod_targets.any?(&:uses_swift?) && deployment_target.to_f < 8.0
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: Needs to check the platform name as well

Comment (neonichu) [inline lib/cocoapods/validator.rb]: :+1: - just randomly works rn because 10.x > 8.0

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -310,6 +310,10 @@ def install_pod
       installer.install!
 
       file_accessors = installer.aggregate_targets.map do |target|
+        if target.pod_targets.any?(&:uses_swift?) && consumer.platform_name == :ios && deployment_target.to_f < 8.0
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: Also, please compare using `Pod::Version` ?

**Pull request merged**